init

```diff
@@ -0,0 +1,455 @@
+            $results[$measureType]['average']
+                = round($this->calculateAverage($$measureType), 4);
+
+            sort($$measureType);
+
+            $results[$measureType]['median']
+                = round($this->calculateMedian($$measureType), 4);
+
+            $results[$measureType]['95p']
+                = round($this->calculate95Percentile($$measureType), 4);
+        }
+
+        $results['cpu']['average'] = round($results['cpu']['average'] / 100, 4);
+        $results['cpu']['median'] = round($results['cpu']['median'] / 100, 4);
+        $results['cpu']['95p'] = round($results['cpu']['95p'] / 100, 4);
+        $results['disc']['read']['total'] = round($discRead / 1024 / 1024, 4);
+        $results['disc']['write']['total'] = round($discWrite / 1024 / 1024, 4);
+        return $results;
+    }
+
+    private function calculateCpuLoad(
+        int $usageUsecStart,
+        int $usageUsecEnd,
+        int $elapsedTimeUsec
+    ): float {
+        $usageMicroseconds = ($usageUsecEnd - $usageUsecStart);
+        $cpuPercentage = ($usageMicroseconds / $elapsedTimeUsec) * 100;
+        return $cpuPercentage;
+    }
+
+
+    private function saveStats(float $startTime): void
+    {
+        $elapsed = microtime(true) - $startTime;
+
+        $date = date('ymd_his');
+        $folderPath = "../../results/$this->testName/$this->engineName/$date/";
+        $fileName = $this->testName . "_" . $this->engineName . "__init";
+
+        if (!file_exists($folderPath)) {
+            mkdir($folderPath);
+        }
+
+        file_put_contents(
+            $folderPath . $fileName,
+            serialize([
+                'stage' => 'init',
+                'elapsedTime' => $elapsed,
+                'metrics' => $this->calculateStats(),
+                'engine' => $this->engineName,
+                'test' => $this->testName,
+                'type' => $this->type,
+                'formatVersion' => 1,
+                'version' => $this->getVersion()
+            ]));
+
+        unlink($this->measuresLogPath);
+    }
+
+
+    private function getVersion(): string
+    {
+        include_once("../../core/engine.php");
+
+        if ($this->engineName === 'mysql_percona') {
+            include_once("../../plugins/mysql.php");
+        }
+        include_once("../../plugins/$this->engineName.php");
+
+
+        $engineClassName = $this->engineName;
+        /** @var engine $engine */
+        $engine = new $engineClassName($this->type);
+        $info = $engine->getInfo();
+        return $info['version'];
+    }
+
+
+    private function calculateAverage(array $values): float|int
+    {
+        return array_sum($values) / count($values);
+    }
+
+    private function calculateMedian(array $sortedValues): float|int
+    {
+        $middleKey = (int) (sizeof($sortedValues) / 2);
+
+        if ($middleKey > 1) {
+            $lowerBound = $sortedValues[$middleKey - 1];
+            $higherBound = $sortedValues[$middleKey + 1];
```

Comment:
This looks a little bit wrong. For a set of e.g. 3 values (e.g. `1, 3, 10` the median value should be the 2nd (i.e. `3`). In your case it will be `(1st + 3rd) / 2 = 5.5`

Also for the set of odd number of elements, e.g. `1, 2, 3, 4` it should be `(2+3)/2`. In your case it seems it will be `(1+3)/2` or `(2+4)/2`

Consider this instead:

```
private function calculateMedian(array $sortedValues): float|int
{
    $count = count($sortedValues);
    $middleKey = (int) ($count / 2);

    if ($count % 2 === 0) {
        // If even, return the average of the two middle values
        return ($sortedValues[$middleKey - 1] + $sortedValues[$middleKey]) / 2;
    } else {
        // If odd, return the middle value
        return $sortedValues[$middleKey];
    }
}

```